CONTRIBUTION.md

```diff
@@ -0,0 +1,103 @@
+- Detailed steps to reproduce the bug
+    - Ex) `File -> Preferences -> General -> Restore Defaults`
+- What you actually expected to do and why
+    - Ex) `DB4S settings are initalized normally.`
+- System environment, DB4S version, loaded extensions
+    - Ex) `macOS Catalina Version 10.15.6(19G2021), DB4S Version 3.12.0, No Extensions`
+- Attach screenshots or screencasts related to the bug/issue
+    - You can add images can be pasted directly into the GitHub issue,  
+    but screencast can be hosted on streamable.com and the link pasted into the issue.
+- Crash log (If the crash log is known, although this is **NOT** required)
+    - If you have a long crash log, [Pastebin](https://pastebin.com/) can help (Make sure to set expiration to 'never')
+- Project file or database file in use (**NOT** required)
+
+A template from the GitHub issue will help you write a good bug/issue report.
+
+### Suggest enhancement
+Please check the following before suggesting an enhancement.
+
+- Search for existing similar issues
+
+And please follow the below to help members identify the suggestion.
+
+- Use a concise but descriptive title to identify the suggestion
+- Describe the detailed steps for the enhancement you want to propose
+- Attach a simple example of the proposal, already applied projects or screenshots
+
+### Translation
+Project translation can be done through the ['Qt Linguist'](http://doc.qt.io/qt-5/qtlinguist-index.html) program.
+Languages that are already being translated can be translated by modifying the corresponding translation file,
+and translations for new languages can be translated by adding a new entry to the TRANSLATION section of the src/src.pro file and then rebuilding the project.
+
+#### DO NOT translate key names in shortcuts
+Keyboard shortcuts get automatically localized by Qt, and translating key names may indeed break them. They show up on the translation file only for the rare occasions that one has to redirect a shortcut to another one in a specific language. But even then, you should use the English key names to get this working. Theoretically, one could follow Qt conventions for translated key names, and they should work, but the risk to stumble upon Qt bugs or moving conventions is high, so it is better to not translate keys and only remap the shortcut. If you want to preserve the original key sequence (recommended), just leave the translation empty and finished.
+
+## Styleguide
+Styleguides help keep the project's source code looking good and consistent.
+This can be of great help to existing and new contributors.
+
+### CPP source code
+#### Indent
+- 4 spaces(**NOT** tabs) for indentation
+
+#### Declare variables
+- Use lowerCamelCase
+- Declare only one variable per line
+- \* in pointer variable is attached to data type
+    - Ex) `int* ptr`
+- Avoid meaningless names (although these can be used for counter or temporary names)
+
+#### Whitespace
+- Do **NOT** use spaces after keywords
```

Comment:
Does this mean that we have to write `}else{`? I guess that is not the intention, but to avoid a space between a keyword and a parenthesis, like in `if(...)`. I have probably introduced some instances of the space option, because I like it more, but given that most of the code adds no space, I agree to make it always without it from now on.

Reply:
While writing this guidelines, I wrote it by referring to the existing code.
I think this part needs to be modified by collecting the opinions of other members or the longest-time contributors.

and, the code style I intended is as follows.

```cpp
if(condition)
{
    statement
} else {
    statement
}
```

Reply:
@MKleusberg May I get any advice on this matter too?

P.S. If it's ambiguous, we can delete this section.
Personally, this is because I don't think it significantly hurts the readability of the code.

Reply:
@MKleusberg Sorry to bother you. If you have time, can you please comment on this matter? 👀

Reply:
This was foreseen. It highlights the need for 'astyle'(http://astyle.sourceforge.net/). This was one aspect of pull request #2238. I've ignored that, unfortunately, due to other projects but it can be revisited. This process, and application of formatting, can be automated.

Reply:
@lucydodo  No worries. I'm happy to look into this :smile: 
Hmm, "Do not use spaces after keywords" seems a bit ambiguous indeed. Also "int a;" obviously has to have a space after the keyword. Maybe something like "Don't add an extra space before a left parenthesis"? This would catch all of these:
```c++
if(condition)
    do_something();

for(int i=0;i<5;i++)
    do_something();

void some_function() {}
```

Or can somebody find a counterexample? :smile:

Reply:
@MKleusberg That's right. To express it as a keyword here is too comprehensive.
It's a slightly different opinion, but after reading and thinking about the PR comments mentioned by @scottfurry above, on the one hand, I think that these rules may reduce work efficiency for members and contributors.

(Sorry to the members who gave us a lot of comments 😞) It may be one option,
delete contents about the code style and make only the rules that we think are simple and necessary. 
- No tab, Use 4 spaces
- Put the curly brackets on the same line as the keyword.